kai/reactive_codeplanner/agent/dependency_agent/dependency_agent.py

```diff
@@ -199,35 +199,49 @@ def execute(self, ask: AgentRequest) -> AgentResult:
         # no result. In this case we will want the sub agent to try and give us additional information.
         # Today, if we don't have the FQDN then we are going to skip updating for now.
 
-        while fix_gen_attempts < self._retries:
+        all_actions: list[_action] = []
+        while fix_gen_attempts < self._max_retries:
             fix_gen_attempts += 1
 
             fix_gen_response = self._model_provider.invoke(msg)
             llm_response = self.parse_llm_response(fix_gen_response.content)
-            # Break out of the while loop, if we don't have a final answer then we need to retry
-            if llm_response is None or not llm_response.final_answer:
+
+            # if we don't have a final answer, we need to retry, otherwise break
+            if llm_response is not None and llm_response.final_answer:
+                all_actions.extend(llm_response.actions)
                 break
 
-            # We do not believe that we should not continue now we have to continue after running the code that is asked to be run.
-            # The only exception to this rule, is when we actually update the file, that should be handled by the caller.
-            # This happens sometimes that the LLM will stop and wait for more information.
+            # we have to keep the chat going until we get a final answer
```

Comment:
should we have some limit here, say 5-8 round trips? if it is more, then something is not correct, and we need to exit?

Reply:
I set the limit to 3